### ISSUE TITLE
Regression `no-member` with inherited AnnAssign nodes

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#7631]
Regression from #7509
The false-positive is actually caused by an issue in astroid. In particular with https://github.com/PyCQA/astroid/blob/v2.12.11/astroid/nodes/scoped_nodes/scoped_nodes.py#L2570-L2577

Opened this so I don't forget to add the example as regression test once the issue is fixed.

```py
from __future__ import annotations

class Base:
    attr: int | None = None

class Parent(Base):
    attr: int

class Child(Parent):
    attr = 2

    def __init__(self):
        self.attr = self.attr | 4  # false-positive
```
```
test.py:13:20: E1101: Instance of 'Child' has no 'attr' member (no-member)
```

**Comment by mbyrnepr2** [comment]
Thanks for your work. I thought I would share [this discussion](https://github.com/PyCQA/pylint/pull/7586#issuecomment-1272265234) because it sounds like we are moving away from adding these tests to Pylint which relate to astroid fixes; I could have misunderstood of course!

**Comment by Pierre-Sassoulas** [comment]
As I said in the discussion you linked, we need to think about it each time imo. If the pipeline time is such a problem we can separate a lot of things and test them in another repository (``pylint.testutil``, for example is useless for pylint direct users and the first one I would do.) Splitting primers in 3 instead of 2 would also work as it's the bottleneck right now.

**Comment by cdce8p** [comment]
> If the pipeline time is such a problem we can separate a lot of things and test them in another repository (`pylint.testutil`, for example is useless for pylint direct users and the first one I would do.) Splitting primers in 3 instead of 2 would also work as it's the bottleneck right now.

I don't think it's an issue. ~5min is completely fine. We could also use something like https://pypi.org/project/pytest-xdist/ to speed them up if it ever becomes a problem.

**Comment by DanielNoord** [comment]
I was mainly referring to local environments. It is not so much about speeding up CI, but speeding up fixing small regressions locally which you want to test before pushing to Github.
Imo, it is annoying to wait on tests locally that are already tested in the library that actually causes them.

**Comment by Pierre-Sassoulas** [comment]
> Imo, it is annoying to wait on tests locally that are already tested in the library that actually causes them.

Sure, we just need to consider the amount of code that uses the astroid fix in pylint. Sometime there is nothing so a test in astroid is enough, but other time there's a quite a lot of checker code and plumbing that also need to be tested.

**Comment by Pierre-Sassoulas** [comment]
Look like this test was indeed required because it's failing on Windows in pylint and this wasn't caught in astroid :)
